### PR TITLE
upgrade dependencies for vulkan-validationlayers

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **vulkan-validationlayers**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
What's the deal with spirv-tools? My changes are not relevant to it.

**Comment by SpaceIm** [comment]
> What's the deal with spirv-tools? My changes are not relevant to it.

Maybe a regression in conan? It fails for cross-build to Macos M1 only.

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 @jgsogo 
please review

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
I have keep private=True for spirv-tools, as it is really needed.

**Comment by AndreyMlashkin** [comment]
Have I made a mistake in _is_cross_building method or is it not about cross compilation?

**Comment by AndreyMlashkin** [comment]
@SpaceIm @SSE4 @uilianries , please approve

**Review by SSE4 — DISMISSED** [review]

File: recipes/vulkan-validationlayers/all/conanfile.py [hunk 1]
```diff
@@ -57,18 +61,21 @@ def _get_compatible_spirv_tools_version(self):
...
             self.requires("xorg/system")
         if self.options.get_safe("with_wsi_wayland"):
-            self.requires("wayland/1.18.0")
+            self.requires("wayland/1.19.0")
 
     def validate(self):
         if self.options["spirv-tools"].shared:
             raise ConanInvalidConfiguration("vulkan-validationlayers can't depend on shared spirv-tools")
+        if tools.is_apple_os(self.settings.os) and self.settings.arch in ["armv8", "armv8_32", "armv8.3"]:
+            raise ConanInvalidConfiguration("vulkan-validationlayers can't be build with this arch; open a PR to fix")
```
Comment (SpaceIm) [inline recipes/vulkan-validationlayers/all/conanfile.py:78]: What's wrong here? I can cross-build this recipe to armv8 if private=False

File: recipes/vulkan-validationlayers/all/conanfile.py [hunk 2]
```diff
@@ -57,18 +61,21 @@ def _get_compatible_spirv_tools_version(self):
         }.get(str(self.version), False)
 
     def requirements(self):
-        self.requires("spirv-tools/{}".format(self._get_compatible_spirv_tools_version), private=True)
+        #TODO set private=False, once the issue is resolved https://github.com/conan-io/conan/issues/9390
+        self.requires("spirv-tools/{}".format(self._get_compatible_spirv_tools_version), private=not self._is_cross_building())
```
Comment (SpaceIm) [inline recipes/vulkan-validationlayers/all/conanfile.py:65]: you mean set private=True once the issue is resolved.

But why not remove private for the moment?

**Comment by SpaceIm** [comment]
Just revert to https://github.com/conan-io/conan-center-index/pull/7195/commits/807398db7170a0e413e4632abe0f8c788f1fae82, I don't understand your last commits.

File: recipes/vulkan-validationlayers/all/conanfile.py (hunk 2 above)
Comment (AndreyMlashkin) [reply recipes/vulkan-validationlayers/all/conanfile.py:65]: I have diligent-core recipe in my repository, it uses a slightly different spirv-tools. It also overrides the namespace.
If we have it not private, it breaks that recipe.

File: recipes/vulkan-validationlayers/all/conanfile.py (hunk 1 above)
Comment (AndreyMlashkin) [reply recipes/vulkan-validationlayers/all/conanfile.py]: Removed Exception

File: recipes/vulkan-validationlayers/all/conanfile.py [hunk 3]
```diff
@@ -57,14 +61,18 @@ def _get_compatible_spirv_tools_version(self):
         }.get(str(self.version), False)
 
     def requirements(self):
-        self.requires("spirv-tools/{}".format(self._get_compatible_spirv_tools_version), private=True)
+        #TODO set private=False, once the issue is resolved https://github.com/conan-io/conan/issues/9390
+        set_private = not self._is_cross_building()
+        if tools.is_apple_os(self.settings.os) and self.settings.arch in ["armv8", "armv8_32", "armv8.3"]:
+            set_private = False
+        self.requires("spirv-tools/{}".format(self._get_compatible_spirv_tools_version), private=set_private)
```
Comment (SpaceIm) [inline recipes/vulkan-validationlayers/all/conanfile.py:68]: The issue is not about cross-build but 2 profiles, so I guess your detection of 2 profiles is wrong.

Comment (AndreyMlashkin) [reply recipes/vulkan-validationlayers/all/conanfile.py:68]: So, how to detect two profiles?

Comment (SpaceIm) [reply recipes/vulkan-validationlayers/all/conanfile.py:68]:
```suggestion
        # TODO: set private=True, once the issue is resolved https://github.com/conan-io/conan/issues/9390
        self.requires("spirv-tools/{}".format(self._get_compatible_spirv_tools_version), private=not hasattr(self, "settings_build"))
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Review by SpaceIm — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@conan-center-bot
continuous-integration/jenkins/pr-merge — The build of this commit was aborted

why that?
I don't see a failed build...

**Comment by jgsogo** [comment]
In fact, build number 6 was aborted, and build number 7 succeeded. Maybe Github sent again the same commit notification... I'm triggering the build again manually. It should report successful status.

**Comment by jgsogo** [comment]
Now, much better 
image

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**